src/Build.OM.UnitTests/Definition/ProjectItem_Tests.cs

```diff
@@ -754,6 +754,34 @@ public void IncludeWithWildcardShouldNotPreserveUserSlashesInFixedDirectoryPart(
             TestIncludeExclude(projectContents, inputFiles, expectedInclude, includeString, "");
         }
 
+        /// <summary>
+        /// Project getter that renames an item to a wildcard that results in drive enumeration exception.
+        /// </summary>
+        [Fact]
+        public void ProjectGetterResultingInDriveEnumerationException()
+        {
+            using (var env = TestEnvironment.Create())
+            {
+                env.SetEnvironmentVariable("MsBuildCheckWildcardDriveEnumeration", "1");
+                Project project = new Project();
+                Assert.Throws<InvalidProjectFileException>(() => { ProjectItem item = project.AddItem("i", "\\**\\*.log")[0]; });
+            }
+        }
+
+        /// <summary>
+        /// Project getter that renames an item to a wildcard that results in logging a warning for attempted drive enumeration.
+        /// </summary>
+        [Fact]
+        public void ProjectGetterResultingInDriveEnumerationWarning()
+        {
+            using (var env = TestEnvironment.Create())
+            {
+                env.SetEnvironmentVariable("MsBuildCheckWildcardDriveEnumeration", "0");
+                Project project = new Project();
+                ProjectItem item = project.AddItem("i", "\\**\\*.log")[0];
```

Comment:
This test case results in an `UnauthorizedAccessException`. Is it possible to manually repro this?